File: bit-manipulation/hammingWeight.java
/**
 Input: 00000000000000000000000000001011
 Output: 3
 Explanation: 
 The input binary string 00000000000000000000000000001011 has a total of three '1' bits.
 */

public class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {
        int res = 0;
        while (n != 0) {
            res++;
            n &= (n - 1);
        }
        return res;  
    }
}

File: bit-manipulation/hammingWeight.py
class Solution(object):
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        rst = 0
        mask = 1
        for i in range(32):
            if n & mask:
                rst += 1
            mask = mask << 1
        return rst